lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -85,6 +90,39 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         if (!account) {
             throw BrowserAuthError.createNoAccountError();
         }
+        const thumbprint: RequestThumbprint = {
+            clientId: this.config.auth.clientId,
+            authority: request.authority || "",
+            scopes: request.scopes,
+            homeAccountIdentifier: account.homeAccountId
+        };
+        const silentRequestKey = JSON.stringify(thumbprint);
+        let response = this.activeSilentTokenRequests.get(silentRequestKey);
+        if (typeof response === "undefined") {
+            this.logger.verbose("acquireTokenSilent called for the first time, storing active request", request.correlationId);
+            response = this.acquireTokenSilentAsync(request, account)
```

Comment:
Can we use `async/await` with a `try/catch` here instead of using `.then/.catch`?

Reply:
We need to cache the entire promise chain, while keeping the cleanup logic outside of acquireTokenSilentAsync, so try/catch won't work.